### DISCUSSION TITLE
Networkx's touch return False while 2 polygons share some edges

### DISCUSSION THREAD

**Discussion by nmcdtu** [networkx/networkx#7413] [Q&A] [answered]
I want to test if some polygons touch the others to create a map with different colors for different regions, by using [greedy_color](https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.coloring.greedy_color.html#networkx.algorithms.coloring.greedy_color) (with `networkx 3.3`).

Here is a reproducible example:
```python
import geopandas as gpd
import networkx as nx

gdf = gpd.read_file('BlackSea_0.json')

# Create an empty graph
G = nx.Graph()

# Add a node for each polygon
for index, row in gdf.iterrows():
    G.add_node(index, polygon=row['geometry'])

# Test if the 2 polygons touch
G.nodes(data=True)[0]["polygon"].touches(G.nodes(data=True)[1]["polygon"])
```

I attached the screen capture 
![polygons](https://github.com/networkx/networkx/assets/147073732/e43f16ca-5bb3-4a1f-b29b-dfc9bfadf499)
 and the sample data [BlackSea_0.json](https://github.com/networkx/networkx/files/14980103/BlackSea_0.json)
As you see, the 2 polygons share the some edges. 
But the test above returned `False`.

I was wondering if I misunderstood something?
I really appreciate if someone can explain it to me, thank you!

**Comment by dschult** [comment]
It looks like the key function/method here is `.touches` which is not a networkx function/method. I guess that it is a geopandas method.

You can cut to the heart of the issue without using networkx:
```python
import geopandas as gpd

gdf = gpd.read_file('BlackSea_0.json')
p0, p1 = gdf['geometry']
p0.touches(p1)   # returns False
```

**Reply by nmcdtu** [reply]
Thank you for your answer! 
Yes, as you wrote, the test did return `False`.

Should I report it in `geopandas` Github repository?

**Reply by nmcdtu** [reply, marked as answer]
I replaced `touches` with `intersects`, and as far as I see, everything worked as expected!
```python
G.nodes(data=True)[0]["polygon"].intersects(G.nodes(data=True)[1]["polygon"])  # return True
```
Thank you very much!

**Comment by dschult** [comment]
Glad you found a solution. 
I will close this conversation. :)
